File: Frontend/Apple/Cookey/Models/EncryptedSessionEnvelope.swift
import Foundation

struct EncryptedSessionEnvelope: Codable {
    let version: Int
    let algorithm: String
    let ephemeralPublicKey: String
    let nonce: String
    let ciphertext: String
    let capturedAt: Date
    let requestSignature: String?

    enum CodingKeys: String, CodingKey {
        case version
        case algorithm
        case nonce
        case ciphertext
        case ephemeralPublicKey = "ephemeral_public_key"
        case capturedAt = "captured_at"
        case requestSignature = "request_signature"
    }
}

File: Frontend/Android/app/src/main/java/wiki/qaq/cookey/model/EncryptedSessionEnvelope.kt
package wiki.qaq.cookey.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class EncryptedSessionEnvelope(
    val version: Int = 1,
    val algorithm: String = "x25519-xsalsa20poly1305",
    @SerialName("ephemeral_public_key")
    val ephemeralPublicKey: String,
    val nonce: String,
    val ciphertext: String,
    @SerialName("captured_at")
    val capturedAt: String,
    @SerialName("request_signature")
    val requestSignature: String? = null
)
